contracts/SafeProtocolMediator.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+import {ISafeProtocolMediator} from "./interfaces/Mediator.sol";
+import {ISafeProtocolModule} from "./interfaces/Components.sol";
+
+import {ISafe} from "./interfaces/Accounts.sol";
+import {SafeProtocolAction, SafeTransaction, SafeRootAccess} from "./DataTypes.sol";
+import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+
+/**
+ * @title TODO
+ * @notice TODO
+ */
+contract SafeProtocolMediator is ISafeProtocolMediator, Ownable2Step {
+    /**
+     * @notice Mapping of a mapping what stores information about modules that are enabled per Safe.
+     *         address (Safe address) => address (component address) => EnabledMoudleInfo
+     */
+    mapping(address => mapping(address => MoudleAccessInfo)) public enabledComponents;
```

Comment:
Is there a way to list all enabled modules? Should we use a linked list similar to how it is done in the Safe contracts?

Reply:
WIP here: https://github.com/5afe/safe-protocol/pull/12/commits/355ee7c5f9b5e5bbf4cb49a3f7d5d8d1e7d24dc6

I am adding some tests and handle all cases.